### PR TITLE
Add contributor guidelines and templates for issues, pull requests and feature requests

### LINKED ISSUE #158: Add Contribution guideline, Issue and Pull Request Template
We need to document how others can start contributing to the tasking library as well as set some templates for our issues and pull requests to make them more understandable and faster to work on.

Would be nice to also detail how to setup this package for development:

- [ ] Detailing our usage of `pipenv`
- [ ] How one can go about setting up our pre-commit checks

### PR DESCRIPTION
**Pull request by DavisRayM**
Closes #158

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Contribution Guidelines
+
+Welcome! Thank you for showing interest in contributing to this package 🥰🥳
+
+The following information compromises of some guidelines & details important in ensuring you're first contribution is as smooth as possible. Following the guidelines also helps communicate that you respect the time of the maintainers and your intent to help develop this project.
```
Comment (moshthepitt) [inline CONTRIBUTING.md]: Did you mean `comprises`?

Comment (DavisRayM) [reply CONTRIBUTING.md]: Yes, let me change that

**Comment by DavisRayM** [comment]
> Really nice. However, I think the pull request template content should go into the issue template and vice versa.

Yikes, didn't notice that. Changed this.

**Comment by DavisRayM** [comment]
> We probably need to link to contributing.md from the main README

I'll add a link in this PR and open a new issue to update some of our documentation. I've noticed we have a lot of TODO's in most of the documentation.

**Pull request merged**